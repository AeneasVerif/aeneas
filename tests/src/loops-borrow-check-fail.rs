//@ [!borrow-check] skip
//@ [borrow-check] known-failure

// We need to use the general rules for joining the loans
fn loop_reborrow_mut() {
    let mut x = 0;
    let mut px = &mut x;
    while *px > 0 {
        x += 1;
        px = &mut x;
    }
}

// We need to imrpove [prepare_ashared_loans]
fn iter_local_shared_borrow() {
    let mut x = 0;
    let mut p = &x;
    loop {
        p = &(*p);
    }
}
