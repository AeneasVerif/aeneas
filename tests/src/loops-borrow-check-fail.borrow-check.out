[[92mInfo[39m ] Imported: tests/llbc/loops_borrow_check_fail.llbc
[[91mError[39m] Inconsistent projection:
- pe: Expressions.Deref
- v:
Values.VBottom
- ty:
(Types.TRef (Types.RErased, (Types.TLiteral (Values.TInteger Values.I32)),
   Types.RMut))
Source: 'tests/src/loops-borrow-check-fail.rs', lines 7:4-12:1
[[91mError[39m] Inconsistent projection:
- pe: Expressions.Deref
- v:
Values.VBottom
- ty:
(Types.TRef (Types.RErased, (Types.TLiteral (Values.TInteger Values.I32)),
   Types.RShared))
Source: 'tests/src/loops-borrow-check-fail.rs', lines 17:4-21:1
